src/S3/PostObjectV4.php

```diff
@@ -54,46 +54,28 @@ public function __construct(
             'method'  => 'POST',
             'enctype' => 'multipart/form-data'
         ];
+
         // setup basic policy documents
-        if (is_null($expiration)) {
-            $expiration = gmdate('Y-m-d\TH:i:s\Z', strtotime('+1 hours'));
-        }
+        $expiration = date (
+            'Y-m-d\TH:i:s\Z',
+            strtotime(gmdate(self::ISO8601_BASIC) . '+' . $expires .' hours')
+        );
         $policy = [
             'expiration' => $expiration,
             'conditions' => [],
         ];
         // policy condition updates
-        $starts_with = []; // handle start-with conditions differently
-        $start_with_key = '';
-        $pool = ['x-amz-credential', 'x-amz-algorithm', 'x-amz-date'];
         foreach ($options as $option) {
-            foreach ($option as $key => $value) {
-                if (is_int($key)) {
-                    if ($key == 0) {
-                        $starts_with = [];
-                    } else if (empty($start_with_key)) {
-                        $start_with_key = substr($value, 1); //remove $
-                    } else {
-                        if (strlen(trim($value)) >= 1) { // validate 'key'
-                            $this->formInputs[$start_with_key] = $value;
-                        }
-                        $start_with_key = '';
-                    }
-                    $starts_with[] = $value;
-                } else {
-                    if (!empty($starts_with)) {
-                        $policy['conditions'][] = $starts_with;
-                        $starts_with = [];
-                    }
-                    $policy['conditions'][] = [$key => $value];
-                    if (in_array($key, $pool)) {
-                        $this->formInputs[ucwords($key, "-")] = $value;
-                    } else {
-                        $this->formInputs[$key] = $value;
-                    }
+            $policy['conditions'][] = $option;
+            if (sizeof($option) > 2) { // starts_with
+                if ($option[1] == '$key' && strlen(trim($option[2])) >= 1){
+                    $this->formInputs['key'] = $option[2];
                 }
+            } else {
+                $this->formInputs += $option;
```

Comment:
I'm not sure adding these values to the form inputs is the right thing to do. What if it's already in the array supplied by the user? How does S3 respond to duplicate fields?

I would just trust that the user has added what they need to the form inputs and the conditions arguments.